File: spec/java/src/io/kaitai/struct/spec/TestTypeTernaryOpaque.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.TypeTernaryOpaque;
import org.testng.annotations.Test;

import static org.testng.Assert.assertEquals;

public class TestTypeTernaryOpaque extends CommonSpec {
    @Test
    public void testTypeTernaryOpaque() throws Exception {
        TypeTernaryOpaque r = TypeTernaryOpaque.fromFile(SRC_DIR + "term_strz.bin");

        assertEquals(r.dif().s1(), "foo");
        assertEquals(r.dif().s2(), "bar");
        assertEquals(r.dif().s3(), "|baz@");
    }
}

File: spec/rust/tests/test_type_ternary_opaque.rs
use std::fs;
extern crate kaitai;
use self::kaitai::*;
use rust::formats::type_ternary_opaque::*;

#[test]
fn test_type_ternary_opaque() -> KResult<()> {
    let bytes = fs::read("../../src/term_strz.bin").unwrap();
    let _io = BytesReader::from(bytes);
    let r: OptRc<TypeTernaryOpaque> = TypeTernaryOpaque::read_into(&_io, None, None)?;

    assert_eq!(*r.dif()?.s1(), "foo");
    assert_eq!(*r.dif()?.s2(), "bar");
    assert_eq!(*r.dif()?.s3(), "|baz@");
    Ok(())
}
